extensions/github-authentication/src/githubServer.ts

```diff
@@ -16,6 +16,10 @@ import { fetching } from './node/fetch';
 
 const CLIENT_ID = '01ab8ac9400c4e429b23';
 const GITHUB_TOKEN_URL = 'https://vscode.dev/codeExchangeProxyEndpoints/github/login/oauth/access_token';
+
+const TIMED_OUT_ERROR = 'Timed out';
+const CANCELLATION_ERROR = 'Cancelled';
+const USER_CANCELLATION_ERROR = 'User Cancelled';
```

Comment:
Do these strings ever end up in UI somewhere (e.g. an uncaught error) that gets bubbled up into an error modal? If so, could we make these a little more descriptive?

Reply:
They shouldn't. `CANCELLATION_ERROR` is thrown to extensions calling `getSession({ createIfNone: true })`... but the others are totally internal.

It's kinda been this unspoken contract that an extension can handle cancelation accordingly by checking for the error message `Cancelled`. It should be more apparent, but that's out of scope of this PR.

Reply:
I added some comments